lib/adapters/http.js

```diff
@@ -227,6 +227,9 @@ module.exports = function httpAdapter(config) {
       if (config.maxRedirects) {
         options.maxRedirects = config.maxRedirects;
       }
+      if (config.beforeRedirect) {
+        options.beforeRedirect = config.beforeRedirect;
```

Comment:
Doesn't this break proxy support when redirection occurs? 

[The setProxy function](https://github.com/axios/axios/pull/3852/files#diff-586c04c24608b86a074073c5514cc195be5c4c87ddb56da233f6d998ada4159fR40), which is called before hitting this line, sets `options.beforeRedirect` to re-apply proxy configuration after a redirect. This line here happily stamps over it. 

Either there should be something to harmonize the two features (such as registering a function that invokes both the original config beforeRedirect if needed, and the proxy re-configuration next) or the documentation should state that the `config.beforeRedirect` option is incompatible with proxy support (😿) and can introduce proxy redirect issues.